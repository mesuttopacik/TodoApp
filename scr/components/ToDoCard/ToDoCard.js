import {View, Text, FlatList, TouchableOpacity} from 'react-native';
import React from 'react';
import styles from './ToDoCard.style';
import IconCard from '../IconCard';



const ToDoCard = ({todos, onDeltask}) => {
  listItemHandler = ({item, index}) => (
    <TouchableOpacity onPress={() => onDeltask(index)} style={styles.container}>
      <View style={styles.innerContainer}>
        <View style={styles.taskText}>
          <Text style={styles.innerText}>{index + 1} - </Text>
          <Text style={styles.innerText}>{item}</Text>
        </View>
      </View>
      <IconCard name="circle-o" />
    </TouchableOpacity>
  );
  return (
    <View>
      <FlatList
        data={todos}
        renderItem={listItemHandler}
        keyExtractor={todos.index}
      />
    </View>
  );
};

export default ToDoCard;
