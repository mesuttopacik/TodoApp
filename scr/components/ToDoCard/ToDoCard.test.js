import {render} from '@testing-library/react-native';
import ToDoCard from './ToDoCard';
import React from 'react';

test('should match with snapshot', () => {
  const comp = render(<ToDoCard />);
  expect(comp).toMatchSnapshot();
});
