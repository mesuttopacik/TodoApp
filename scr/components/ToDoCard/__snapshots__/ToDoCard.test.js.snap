// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View>
  <RCTScrollView
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="undone-task"
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
</View>
`;
